engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -121,10 +132,33 @@ func (pool *EnclavePool) GetEnclave(
 		return nil, nil
 	}
 
+	// If there is no idle enclave in the pool returns nil
+	// for not to block the caller
+	if len(pool.idleEnclavesChan) < oneIdleEnclave {
```

Comment:
nit: I think you can do `== 0` here, using a const is a bit overkill IMO

Reply:
Returning `nil, nil` usually frightens me. Are we 100% certain this won't cause an NPE downstream? And BTW what will happen in this case? Are we creating an encalve on the fly?

Reply:
I am also not too happy with returning `nil, nil` but after reviewing it several times I didn't find a better cleaner way to solve this use case (one alternative was returning another value for specifying that not enclave was found in the pool, but I though it wasn't to clean), I added a comment about it on the method for this reason. And yes, it's actually handled from the caller's side

<img width="630" alt="image" src="https://github.com/kurtosis-tech/kurtosis/assets/29951623/13daf6fb-b7b9-4c6b-b4d0-95bbb5508d4e">

Let me know if you have any other suggestion I could refactor it in a next PR